Add docs for Multi-Package Daml Studio support (#851)
digital-asset/docs.daml.com: docs/2.10.0/docs/daml/daml-studio.rst

- The ``data-dependencies`` field of the current package or any other packages known to the ``multi-package.yaml``
- Under the added ``dars`` field in the ``multi-package.yaml``, which is shown below:

.. code:: yaml

  packages:
  - my-package
  - libs/my-lib
  dars:
  # Given my-package depends on some DAR, which then depends on my-transitive-dependency
  - ./dars/my-transitive-dependency-1.0.0.dar

Adding DAR paths to this field will have no effect on building, serving only for the IDE to jump to.

Multiple Daml SDK Versions
==========================
Separate package environments can run in different Daml SDK versions (though some features may
not work correctly in versions <= 2.8). When opening a file, or jumping to a location
in a package running a Daml version that is not currently installed, Daml Studio will prompt
you to install the missing Daml SDK, and handle the installation, progress reporting and
cancellation within a notification.

Directory Envrionment Tools (direnv)
====================================

> It is common to use tools like ``direnv`` to set up dependencies and import environment variables
> for use with :ref:`environment variable interpolation support <environment-variable-interpolation>`. In order to have this working in Daml Studio,
> you will need a VSCode extension that sets this up for other extensions.

Reviewer: ```suggestion
Tools like ``direnv`` are commonly used to set up dependencies and import environment variables
for use with :ref:`environment variable interpolation support <environment-variable-interpolation>`. To make this work in Daml Studio,
you need a VSCode extension that sets this up for other extensions.
```

In the case of ``direnv`` specifically (i.e. you have a ``.envrc`` file), we
recommend using `this direnv extension by Martin Kühl <https://marketplace.visualstudio.com/items?itemName=mkhl.direnv>`__, which we have verified is compatible.
Other direnv extensions may not correctly pass environment information to the Daml Studio extension.

> If the Daml extension detects a ``.envrc`` file, it will recommend this extension within the IDE with the
> following message:

Reviewer: ```suggestion
If the Daml extension detects a ``.envrc`` file, it recommends this extension within the IDE with the
following message:
```

..code::

  Found an .envrc file but the recommended direnv VSCode extension is not installed. Daml IDE may fail to start due to missing environment variables.
  Would you like to install the recommended direnv extension or attempt to continue without it?

> It will also provide a link to the extension on the VS Code extension marketplace.

Reviewer: ```suggestion
It also provides a link to the extension on the VS Code extension marketplace.
```

Limitations in 2.10.0-rc
========================

> These limtations are for the Release Candidate only, and will be fixed for 2.10 full release.

Reviewer: ```suggestion
These limitations are for the Release Candidate only, not for the 2.10 full release.
```

* Jumping to non-local dependencies does not currently retain the build-options and module-prefixes
  for that package. This means that if you jump to a dependency that requires either of these to build,

>   the editor will show errors on the source code.

Reviewer: ```suggestion
  the editor shows errors in the source code.
```

* Some links in the Script Results tab may not resolve correctly cross-package.
* Packages with symlinks between the daml.yaml and source files may not give correct intelligence.

Setting Multi-IDE SDK Version
=============================

> The Multi-IDE will select the correct SDK version for each package you interact with, however the
> Multi-IDE itself will run from your most recent installed SDK version. You can override this using
> an additional ``daml.yaml`` file at the root of your project (next to the ``multi-package.yaml``)
> containing only an SDK Version, as follows:

Reviewer: ```suggestion
The Multi-IDE selects the correct SDK version for each package you interact with. However the
Multi-IDE itself runs from your most recently installed SDK version. You can override this using
an additional ``daml.yaml`` file at the root of your project (next to the ``multi-package.yaml``)
containing only an SDK Version, as follows:
```

any packages listed in a top level ``multi-package.yaml`` file.

Multi-package.yaml Location
===========================
The ``multi-package.yaml`` file will only be used by Daml Studio if it sits at the
Workspace Root, i.e. in the root directory that Daml Studio was opened in.
If no ``multi-package.yaml`` is found, the cross-package jump-to-definition feature
will not jump to the real on-disk source code, however other Multi-IDE features will
still work as normal.

Package Environments and Hot-Loading
====================================
The Multi-IDE runs separate package environments for each package in the project,
allowing the IDE to replicate properties such as dependencies, module-prefixes and build-options
of each package individually, without clashing with other packages in the project.
These enrivonments will automatically reload if the relevant ``daml.yaml`` file changes,
or any of the dependency DAR files change.  

Note, however, that the Multi-IDE does not hold "live" copies of DARs, therefore any changes
to a package that affects another package will not be visible in that second package until

> running ``daml build``.  

Author: The multi-ide does this for you :)

        │   └── B.daml
        └── daml.yaml

The root ``daml.yaml`` contains only the ``sdk-version`` field, as shown below.
Note that Multi-IDE is only supported in 2.10+.

.. code-block:: yaml

  sdk-version: 2.10.0

> This feature can also be used when Multi-IDE is disabled, as below.

Reviewer: ```suggestion
This feature can also be used when Multi-IDE is inactive, as described below.
```

> Disabling Multi-IDE
> ===================

Reviewer: ```suggestion
Deactivating Multi-IDE
======================
```

> The Multi-IDE feature can be disabled by going to VSCode settings (``ctrl + ,``), searching for 
> ``Daml: Multi Package Ide Support``, and unticking the box. You do not need to restart your editor.
> If you are disabling this feature due to a bug, please report it on the `forums <https://discuss.daml.com/>`__.

Reviewer: ```suggestion
The Multi-IDE feature can be deactivated by going to VSCode settings (``ctrl + ,``), searching for 
``Daml: Multi Package Ide Support``, and unticking the box. You do not need to restart your editor.
If you are deactivating this feature due to a bug, please report it on the `forums <https://discuss.daml.com/>`__.
```
